src/main/scad/laszlo-molnar/no-holes-barred/molnar.no-holes-barred.scad

```diff
@@ -0,0 +1,68 @@
+include <puzzlecad.scad>
+
+box_puzzle_border = 6;
+box_puzzle_top_inset = 1;
+
+$burr_inset = 0.12;
+$burr_bevel = 1;
+$unit_beveled = true;
+$burr_scale = 17;
+
+dim = $burr_scale * 3 + box_puzzle_border * 2 + $burr_inset * 2;
+height = $burr_scale * 3 + box_puzzle_border + $burr_inset;
+
+*box();
+*pieces();
+*piece_triangle();
+
+module box() {
+    render(convexity = 2)
+    difference() {
+        beveled_cube([dim, dim, height + box_puzzle_top_inset]);
+        translate([box_puzzle_border, box_puzzle_border, box_puzzle_border])
+        cube([
+            $burr_scale * 3 + $burr_inset * 2,
+            $burr_scale * 3 + $burr_inset * 2,
+            $burr_scale * 4
+        ]);
+         
+        translate([dim / 2 - $burr_scale / 4, box_puzzle_border, height - $burr_scale / 4])
+        rotate(a=90, v=[1, 0, 0])
+        cylinder(r = 2.2, h = 2.2, $fn = 32);
+        
+        translate([dim / 2 + $burr_scale / 4, box_puzzle_border, height - $burr_scale / 4])
+        rotate(a=90, v=[1, 0, 0])
+        cylinder(r = 2.2, h = 2.2, $fn = 32);
+        
+        translate([dim / 2, box_puzzle_border, height - $burr_scale * 3/ 4])
+        rotate(a=90, v=[1, 0, 0])
+        cylinder(r = 2.2, h = 2.2, $fn = 32);
+     }
+}
+
+module pieces() {
+    burr_plate([
+        ["xxx|x..", "...|x.."],
+        ["xxx|x..", "...|x.."],
+        ["xxx|x..", "...|x.."],
+        ["xxx|x..", "...|x.."],
+        ["xxx{components={x-,y-,z+y-,z+x-,z-y-,z-x-}}|x.."]
+    ], $plate_width = 160);
+}
+
+module piece_triangle() {
+   burr_plate([
+       ["x{components={x-,y-,z+y-,z+x-,z-y-,z-x-}}"]
+   ]);
+   translate([$burr_scale * 1 / 4, 0, $burr_scale * 1 / 4])
```

Comment:
I'd suggest putting 0.001 mm of overlap here, such as:

translate([$burr_scale * 1 / 4, 0.001, $burr_scale * 1 / 4])

Generally good practice to ensure the final shape is a manifold (& avoid floating point errors).

I'd also suggest putting the "male" alignment pins on the box, and the "female" ones on the piece_triangle... the reason is that on some printers/materials, the indentations on the inside of the box might cause the outside surface of the box to appear slightly uneven. Probably not an issue with box_border set to 6, but might be an issue with thinner borders. (Just a thought)

Reply:
💯